ecc/internal/gpoint/scalarmul.go

```diff
@@ -154,96 +183,154 @@ func (p *{{.Name}}Jac) MultiExpNew(curve *Curve, points []{{ .Name}}Affine, scal
 		}()
 		return chRes
 	}
-	// if we have m points and n cpus, m/n points per cpu
-	nbCpus := runtime.NumCPU()
-	// nb points processed by one cpu
-	pointsPerCPU := len(scalars) / nbCpus
-	// each cpu has its own bucket
-	sharedBuckets := make([][32][255]{{.Name}}Jac, nbCpus)
-	// bucket to gather cpus work
-	var commonBucket [32][255]{{.Name}}Jac
-	var emptyBucket [255]{{.Name}}Jac
-	var almostThere [32]{{.Name}}Jac
-	for i := 0; i < 255; i++ {
-		emptyBucket[i].Set(&curve.{{toLower .Name}}Infinity)
+
+	// var nbChunksPerLimbs, nbChunks, chunkSize int
+	var nbChunks, chunkSize int
+	var mask uint64
+	if len(scalars) <= 10000 {
+		chunkSize = 8
+	} else if len(scalars) <= 80000 {
+		chunkSize = 11
+	} else if len(scalars) <= 400000 {
+		chunkSize = 13
+	} else if len(scalars) <= 800000 {
+		chunkSize = 14
+	} else {
+		chunkSize = 16
 	}
-	
-	const mask = 255
-	// id: cpu id, start, end: point nb start to point nb end, chunk: i-th digit (in corresponding basis)
-	worker := func(id, start, end int) {
-		// ...[chunk*8..(chunk+1)*8-1]... -th bits
-		for chunk := 0; chunk < 32; chunk++ {
-			limb := chunk / 8
-			offset := (chunk % 8) * 8
-			sharedBuckets[id][chunk] = emptyBucket
-			var index uint64
-			for i := start; i < end; i++ {
-				index = (scalars[i][limb] >> offset)
-				index &= mask
-				if index != 0 {
-					sharedBuckets[id][chunk][index-1].AddMixed(&points[i])
-				}
+
+	var bitsForTask [][]int
+	if 256%chunkSize == 0 {
```

Comment:
too many magics number here --> 255, 256 etc should be constant referencing fr.Element defined constants